src/IceRpc.Slice.Generators/Internal/Emitter.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) ZeroC, Inc.
+
+using System.Runtime.InteropServices;
+
+namespace IceRpc.Slice.Generators.Internal;
+
+internal class Emitter
+{
+    private static readonly string _newLine = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "\r\n" : "\n";
+
+    internal string Emit(IReadOnlyList<ServiceClass> serviceClasses, CancellationToken cancellationToken)
+    {
+        string generated = "";
+        foreach (ServiceClass serviceClass in serviceClasses)
+        {
+            // stop if we're asked to.
+            cancellationToken.ThrowIfCancellationRequested();
+
+            string methodModifier =
+                serviceClass.HasBaseServiceClass ? "public override" :
+                serviceClass.IsSealed ? "public" : "public virtual";
+
+            string dispatcherBlock;
+            if (serviceClass.ServiceMethods.Count > 0)
+            {
+                string dispatchBlocks = "";
+                foreach (ServiceMethod serviceMethod in serviceClass.ServiceMethods)
+                {
+                    dispatchBlocks += @$"
+
```

Comment:
Is the goal of this extra space to provide spacing between each case statement? Maybe we could use a list instead to hold the blocks, then trim them and join with a double. newline separator later when added to `dispatcherBlock`?

Reply:
Yes, this provides the spacing between each case statement, I will check if using a list simplifies things.

Reply:
Reworked to use a List